analyzers/src/SonarAnalyzer.CSharp/Rules/SillyMathematicalComparison.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class SillyMathematicalComparison : SonarDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S2198";
+        private const string MessageFormat = "Comparison to integral constant is useless; the constant is outside the range of type '{0}'";
+
+        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterNodeAction(
+                CheckBinary,
+                SyntaxKind.GreaterThanExpression,
+                SyntaxKind.GreaterThanOrEqualExpression,
+                SyntaxKind.LessThanExpression,
+                SyntaxKind.LessThanOrEqualExpression);
+
+        private static void CheckBinary(SonarSyntaxNodeReportingContext context)
+        {
+            var binary = (BinaryExpressionSyntax)context.Node;
+            var typeIdentifier = string.Empty;
+            if (ShouldRaise(context.SemanticModel, binary.Left, binary.Right, ref typeIdentifier))
+            {
+                context.ReportIssue(Diagnostic.Create(Rule, binary.GetLocation(), typeIdentifier));
```

Comment:
Roslyn provides an API for naming things in the shortest possible way:
```
var typeName = context.SemanticModel.GetSymbolInfo(other).Symbol.ToMinimalDisplayString(model, other.GetLocation().SourceSpan.Start)
```

Reply:
I tried it, but for some reason it gives me the name of the entire node, for example `ulong variableName`.

I ended up using this, which is also slightly more succinct:
```csharp
// returns ulong, instead of 'ulong variableName'
var typeName = context.SemanticModel.GetTypeInfo(second).Type.ToString(); 
```

It also works if you declare the type as `System.UInt64`, giving back the simplest form of the type (the primitive `ulong`).

Note: Included the `ulong` and `long` types, since they are not checked by the `CS0652` compiler warning.

Reply:
You probably called it on an `ILocalSymbol` which represents a local variable. You need it for the ITypeSymbol of the type of the expression. Use `semanticModel.GetTypeInfo(second)` and call ToDisplayString on it.